lib/BlockedBy.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const NodeResque = require('node-resque');
+const { runningJobsPrefix } = require('../config/redis');
+
+class BlockedBy extends NodeResque.Plugin {
+    /**
+     * Checks if there are any blocking jobs running. If yes, re-enqueue
+     * If no, set the current job ID as key
+     * @method beforePerform
+     * @return {Promise}
+     */
+    async beforePerform() {
+        const { jobId, blockedBy } = this.args[0];
+        const blockingJobKeys = await this.queueObject.connection.redis.mget(blockedBy);
+        const blockingJobsRunning = blockingJobKeys.some(j => j !== null);
+        const key = `${runningJobsPrefix}${jobId}`;
+
+        // If any blocking job is running, then re-enqueue
+        if (blockingJobsRunning) {
+            await this.reEnqueue();
+
+            return false;
+        }
+
+        // Register the curent job as running by setting key
+        // Set expire time to take care of the case where
+        // afterPerform failed to call and blocked jobs will be stuck forever
+        await this.queueObject.connection.redis.set(key, true);
```

Comment:
Would be easier to maintain a single set with jobs and SADD/SREM off the set in afterPerform(), but I guess you can't set TTL on a set item.

Reply:
Actually, just set the value to an empty string "" instead of "true". Saves space (lesser bitcount https://redis.io/commands/bitcount) since we don't care about the value anyway.

Reply:
Sure. I can do that